CHANGELOG.md

```diff
@@ -87,6 +87,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 * Description tempfiles created via `jj describe` now have the file extension
```

Comment:
Re the commit message (for the last commit):

I might be missing something, since I'm not sure I fully understood the connection between the default value and unifying revsets and revset-aliases yet.

However, naively, the following seems to make more sense:

- Default `revsets.short-prefixes` to "" in rust code.
- In TOML, set its default value.
 
I would still prefer `(remote_branches(main)|remote_branches(master)|remote_branches(develop)|tags())..` as the default value.

For `revset.log`, I would actually make the default to be `all()` in Rust and something else in TOML.

As an aside, perhaps the noop value for `revsets.short-prefixes` should be `none()`?

Reply:
Regarding the default for `revsets.short-prefixes`, one motivation is that it's pretty weird to see the revsets grow when you do `jj git push --all`.

My suggested default for short-prefixes might also be a decent default for log (though it should also include the parents of those commits).

I don't mean to block the PR on figuring out the defaults, though.

-----------------------------------

Also, here's a weird behavior I'm seeing with the default setting, right after `jj git push --all`. Why does `V` get a short id? (If it's not obvious to you, I'll try to get more detailed debug info).

![image](https://github.com/martinvonz/jj/assets/4123047/9ef93c77-e5d7-4df5-9b37-0a3d61653394)

Reply:
> Why does V get a short id?

It looks like it's the "base" for your `X` commit. It should show up in plain `jj log` for the same reason.

Reply:
> Re the commit message (for the last commit):
> 
> I might be missing something, since I'm not sure I fully understood the connection between the default value and unifying revsets and revset-aliases yet.

What I meant was that if `[revset-aliases]` gets merged into `[revsets]`, then the user can define `revsets.short-prefixes = "log"` and `revsets.log = "..."`, but if they're not merged, then the user cannot do `revsets.short-prefixes = "log"`.

> However, naively, the following seems to make more sense:
> 
> * Default `revsets.short-prefixes` to "" in rust code.
> * In TOML, set its default value.

Yes, it might be a little cleaner to put the defaults in TOML so most defaults are collected in one place, and so they show up in `jj config list --include-defaults`.

> I would still prefer `(remote_branches(main)|remote_branches(master)|remote_branches(develop)|tags())..` as the default value.

Do you mean without the "base" commits or did you mean to include them? I may I misunderstood you on Discord because I thought you said you want short hashes for the base commits too.

> For `revset.log`, I would actually make the default to be `all()` in Rust and something else in the default TOML config.
> 
> As an aside, perhaps the noop value for `revsets.short-prefixes` should be `none()`?

I agree that we should move configs to TOML where possible.

Reply:
> Do you mean without the "base" commits or did you mean to include them? 

I don't feel very strongly about it, but I would prefer to include the "base" commits in the `short-prefixes` default. I just forgot to mention them.

Including "base" commits seems more useful in the `log` default. But then, there are more changes I'd like to make to the `log` default. For example, I think it'd also be nice for that to include the tips of local branches.

Reply:
I'll leave this for later to avoid blocking this PR on what seems like a larger discussion. I can't even move the default value to TOML in this PR since the default is "same as revsets.log", which can't be expressed in TOML unless we merge `[revsets]` and `[revset-aliases]`.